test/unit/android-helper-specs.js

```diff
@@ -129,21 +129,45 @@ describe('Android Helpers', () => {
   }));
 
   describe('getDeviceInfoFromCaps', () => {
+    // list of device/emu udids to their os versions
+    // using list instead of map to preserve order
+    let devices = [
+      {udid: 'emulator-1234', os: '4.9.2'},
+      {udid: 'rotalume-1339', os: '5.1.5'},
+      {udid: 'rotalume-1338', os: '5.0.1'},
+      {udid: 'rotalume-1337', os: '5.0.1'},
+      {udid: 'roamulet-9000', os: '6.0'},
+      {udid: 'roamulet-0', os: '2.3'},
+      {udid: '0123456789', os: 'wellhellothere'}
+    ];
+    let curDeviceId = '';
+
     before(() => {
       sinon.stub(ADB, 'createADB', async () => {
         return {
           getDevicesWithRetry: async () => {
-            return [
-              {udid: 'emulator-1234'},
-              {udid: 'rotalume-1337'}
-            ];
+            let udids = [];
```

Comment:
lodash has some handy functions for this sort of thing.

The equivalent would be

```js
return _.map(devices, (device) => { return {udid: device.udid}; });
```